### ISSUE TITLE
Mobile - Screen Share :- Starting/Stopping screen share seems to be invoking onRemoteAudioStateChanged by itself

### ISSUE THREAD

**Issue by bhavinb98** [AgoraIO-Extensions/Agora-Flutter-SDK#841]
This is the method I'm calling. Video is muted before starting the screen share and the audio publishing depends on whether the sharer has muted themselves or not.

```
void startScreenShare() async {
    await _engine.startScreenCapture(
        const ScreenCaptureParameters2(captureAudio: true, captureVideo: true));
    
    await _engine.updateChannelMediaOptionsEx(
      options: ChannelMediaOptions(
        publishMicrophoneTrack: muted,
        publishScreenTrack: true,
        publishSecondaryScreenTrack: true,
        publishScreenCaptureAudio: true,
        publishScreenCaptureVideo: true,
        clientRoleType: ClientRoleType.clientRoleBroadcaster,
      ),
      connection:
          RtcConnection(channelId: channelId, localUid: user!.id!),
    );
  }
```
The stop method.

```
void stopScreenShare() async {
    await _engine.stopScreenCapture();
  }
```

Logs of the `onRemoteAudioStateChanged ` after starting screen share, User had kept themselves muted when starting screen share: - 

```
I/flutter ( 4165): [ APPDEBUG ] 2022-12-15 20:59:45.195881 - REASON - RemoteAudioStateReason.remoteAudioReasonRemoteUnmuted
I/flutter ( 4165): [ APPDEBUG ] 2022-12-15 20:59:45.201252 - REASON - RemoteAudioStateReason.remoteAudioReasonRemoteUnmuted
```

Logs of the `onRemoteAudioStateChanged ` after stopping screen share, User had also kept themselves muted when stopping screen share: - 

```
I/flutter ( 4165): [ APPDEBUG ] 2022-12-15 21:03:13.955909 - REASON - RemoteAudioStateReason.remoteAudioReasonRemoteMuted
I/flutter ( 4165): [ APPDEBUG ] 2022-12-15 21:03:16.155073 - REASON - RemoteAudioStateReason.remoteAudioReasonRemoteUnmuted
I/flutter ( 4165): [ APPDEBUG ] 2022-12-15 21:03:16.157301 - REASON - RemoteAudioStateReason.remoteAudioReasonRemoteUnmuted
I/flutter ( 4165): [ APPDEBUG ] 2022-12-15 21:03:17.956715 - REASON - RemoteAudioStateReason.remoteAudioReasonRemoteMuted
I/flutter ( 4165): [ APPDEBUG ] 2022-12-15 21:03:19.158095 - REASON - RemoteAudioStateReason.remoteAudioReasonRemoteUnmuted
```
The amount of time it is called is random. Sometime there are double the amount of logs generated.

flutter agora version: 6.0.0

**Comment by littleGnAl** [comment]
I think it's expected if in the live-streaming mode
https://api-ref.agora.io/en/voice-sdk/flutter/6.x/API/class_irtcengineeventhandler.html#callback_irtcengineeventhandler_onremoteaudiostatechanged

**Comment by bhavinb98** [comment]
Hmm, but if the user has explicitly muted themselves, and even setting `publishMicrophoneTrack: false` in `ChannelMediaOptions` while starting screen share shouldn't turn it on right?

Another thing I noticed was after stopping screen share, going to back to the logs I sent above, always ends with this - RemoteAudioStateReason.remoteAudioReasonRemoteUnmuted

```
I/flutter ( 4165): [ APPDEBUG ] 2022-12-15 21:03:19.158095 - REASON - RemoteAudioStateReason.remoteAudioReasonRemoteUnmuted
```

**Comment by bhavinb98** [comment]
After further inspection, I found that it also affects video streams.

Anyways, it seems I have found the solution.

https://github.com/AgoraIO-Extensions/Agora-Flutter-SDK/blob/84fdfc275bf31d387cbb18b6fe470e43f0a7cf7f/example/lib/examples/advanced/screen_sharing/screen_sharing.dart#L366

After invoking the `_rtcEngine.stopScreenCapture`, the channel media options should be updated again which solves the issue.
Setting `publishMicrophoneTrack:true` and ` publishScreenTrack: true`, while everything else will be false. 


```
options: ChannelMediaOptions(
        publishMicrophoneTrack: true,
        publishScreenTrack: true,
        publishSecondaryScreenTrack: false,
        publishScreenCaptureAudio: false,
        publishScreenCaptureVideo: false,
        clientRoleType: ClientRoleType.clientRoleBroadcaster,
      ),
```

Vice versa in for starting screen share.
